test/browser_tests/conf.js

```diff
@@ -25,10 +25,14 @@ function _chooseSuite( params, capabilities ) {
   // set the capabilities to use the essential suite.
   // This way setting the browser, for instance, won't run several
   // concurrent times.
-  if ( _paramIsSet( params.browserName ) ||
-       _paramIsSet( params.version ) ||
-       _paramIsSet( params.platform ) ) {
-    capabilities = defaultSuites.essential;
+  capabilities = defaultSuites.essential;
+  if ( !_paramIsSet( params.browserName ) &&
+       !_paramIsSet( params.version ) &&
+       !_paramIsSet( params.platform ) &&
+       ( typeof params.sauce === 'undefined' ||
+                params.sauce === 'true' ) &&
+       _isSauceCredentialSet() ) {
+    capabilities = defaultSuites.full;
```

Comment:
Can this be simplified? Maybe

`var paramsAreNotSet =(  _paramIsSet( params.browserName ) &&
                                         _paramIsSet( params.version ) && 
                                        _paramIsSet( params.platform ) ) ===  false;`

`var useSauceCredentials = ( typeof params.sauce === 'undefined' ||  params.sauce === 'true' ) && _isSauceCredentialSet() `

`If( paramsAreNotSet && useSauceCredentials )...`